src/pages/robot-list/index.js

```diff
@@ -0,0 +1,182 @@
+      const filterType = typeOrder[0];
+      const isAscending = typeOrder[1] === "up" ? 1 : -1;
+
+      const sortedArray = [...filteredRobotList].sort((a, b) =>
+        a[filterType] > b[filterType] ? isAscending : -isAscending
+      );
+
+      setRobotList(sortedArray);
+    };
+    orderArrayBy(sortOptionType);
+  }, [sortOptionType]);
+
+  function buttonOptions(robotIndex, option, pageNumber) {
+    const correctedIndex = robotIndex + (pageNumber - 1) * maxEntriesPerPage;
+    setDeleteRobotIndex(correctedIndex);
+    setExtraRobotInformationIndex(correctedIndex);
+    if (option === "delete") {
+      setShowDeleteModal(true);
+    } else if (option === "information") {
+      setShowRobotModal(true);
+    }
+  }
+
+  function deleteItem() {
+    setRobotList(robotList.filter((item, index) => index !== deleteRobotIndex));
+    setShowDeleteModal(false);
+  }
+
+  const getPaginatedData = (pageNumber) => {
+    const startIndex = pageNumber * maxEntriesPerPage - maxEntriesPerPage;
+    const endIndex = startIndex + maxEntriesPerPage;
+    return filteredRobotList.slice(startIndex, endIndex);
+  };
+
+  function RobotLoadList() {
+    let { pageNumber } = useParams();
+
+    return [
+      <LoadList
+        Robots={getPaginatedData(pageNumber)}
+        onButton={buttonOptions}
+      />,
+      <Pagination maxPage={maxPages} pageLimit={pagesToShow} />,
+    ];
+  }
+
+  return (
+    <div>
+      {
+        /* Only render modal if items are loaded */
+        // main takeaway for this PR is, name your variables more specifically, helps with debugging
+        robotList.length > 0 ? (
+          [
+            showDeleteModal && (
+              <DeleteModal
+                onDo={deleteItem}
+                onClose={() => setShowDeleteModal(false)}
+                RobotNameToBeDeleted={robotList[deleteRobotIndex].name}
+              />
+            ),
+            showRobotModal && (
+              <RobotModal
+                onClose={() => setShowRobotModal(false)}
+                currentRobot={robotList[extraRobotInformationIndex]}
+              />
+            ),
+          ]
+        ) : (
+          <p>Loading ...</p>
+        )
+      }
+      <div>
+        <header className="robot-list-main-header">
+          <Search searchQuery={searchQuery} setSearchQuery={setSearchQuery} />
+        </header>
+      </div>
+      <Switch>
+        <main>
+          <h1 style={{ textAlign: "center" }}>RoboFriends</h1>
+          
+          <div className="robot-list-main-center">
+          <Link to="/">
+            <button className="robot-list-center-link">Home</button>
+          </Link>
+            <SortByDropDown setSortOptionType={setSortOptionType} />
+          </div>
+          {isListLoaded ? (
+            <div className="robot-list-main-container">
+              {error ? (
+                <div>Error: {error.message}</div>
+              ) : (
+                <>
+                  <Route
```

Comment:
Related to my earlier comment about your route name, you should never have a nested route within a `<Route>`.
Since this file is within a `<Route>` in `App.js`, you should not have `<Route>` here.
Instead, think of a way to render the list without wrapping it in `Route`